source/Ship.cpp

```diff
@@ -1621,6 +1627,11 @@ void Ship::DoGeneration()
 			heat += (hullAvailable - hullRemaining) * hullHeat;
 		if(shieldsAvailable)
 			heat += (shieldsAvailable - shieldsRemaining) * shieldsHeat;
+		
+		// Decrease the shield and hull delays by 1 now that shield generation
+		// and hull repair have been skipped over.
+		shieldDelay = max(0., shieldDelay - 1.);
```

Comment:
Note that once a ship gets disabled, `DoGeneration` doesn't get executed. Is it intentional that these delays would then only kick in once the ship gets repaired?

Reply:
Yes, that is intentional. This is actually necessary for the disabled hull delay to work given the current point that I’m setting that delay, as otherwise the disabled hull delay would begin decreasing while the ship is disabled instead of right when the ship becomes enabled again.